src/server/src/main/java/io/cassandrareaper/core/RepairSegment.java

```diff
@@ -172,29 +172,26 @@ public Builder startTime(@Nullable DateTime startTime) {
     }
 
     public Builder endTime(DateTime endTime) {
-      Preconditions.checkNotNull(endTime);
       this.endTime = endTime;
       return this;
     }
 
     public RepairSegment build(@Nullable UUID segmentId) {
-      // a null segmentId is a special case where the storage uses a sequence for it
-      Preconditions.checkNotNull(runId);
-      //Preconditions.checkState(null != startTime || null == endTime, "if endTime is set, so must startTime be set");
-      //Preconditions.checkState(null == endTime || State.DONE == state, "endTime can only be set if segment is DONE");
-      /* Preconditions.checkState(
-      null != startTime || State.NOT_STARTED == state,
-      "startTime must be set if segment is RUNNING or DONE"); */
-
-      if (null != endTime && State.DONE != state) {
-        endTime = null;
-      }
-
-      if (startTime == null && endTime != null) {
-        startTime = endTime.minusSeconds(1);
-      }
-
```

Comment:
I would want to put the preconditions back in place now.
They are there to check our design, and prevent regressions.
I see the clash with how this PR is coded though… need some further thought on this…

Reply:
i took a stab at this, looking for a way to do the same thing with less code and came up with the following…
 https://github.com/thelastpickle/cassandra-reaper/commit/45e67f2aadad55baa937f8fbe7ae4aea9aefac9a

this is a smaller patch, although CassandraStorage still needs a way to re-null the endTime, and there's probably some other small things required, but it demonstrates i think a simpler way.

that said: having coded up an alternative; i've become less at ease with the approach.
my initial feeling here is that we're making the code more complicated, rather than directly solving the problem.

i'd like to put together a PR that a) directly solves the problem and b) provides a way to disable the precondition checks via the use of a system property. hope this sounds ok, then we can compare approaches and re-evaluate…